torch_glow/src/PyTorchFileLoader.cpp

```diff
@@ -15,30 +15,188 @@
+  try {
+    module->forward(inputs);
+  } catch (const std::exception &x) {
+    RETURN_ERR(x.what());
+  }
+  return llvm::Error::success();
+}
+
+/// Helper struct, which on constructor registers custom fusion pass
+/// and operator.
+struct RegisterCustomFusionPass {
+  RegisterCustomFusionPass() {
+    auto options = c10::OperatorOptions();
+    options.setAliasAnalysis(at::AliasAnalysisKind::PURE);
+    torch::jit::RegisterOperators op({torch::jit::Operator(
+        getFusionSymbol(),
+        [](const torch::jit::Node *node) {
+          return [node](torch::jit::Stack &stack) {
+            // Get JIT Graph.
+            auto graph = node->g(at::attr::Subgraph);
+            auto err = convertJitGraphToGlowFunction(
+                stack, *graph, *localFusionInfo.function,
+                *localFusionInfo.inputPlaceholders,
+                *localFusionInfo.outputPlaceholders);
+            if (static_cast<bool>(err)) {
+              throw std::invalid_argument(llvm::toString(std::move(err)));
+            }
+            return 0;
+          };
+        },
+        options)});
+
+    torch::jit::RegisterPass pass([&](std::shared_ptr<torch::jit::Graph> &g) {
+      // Trigger custom fusion pass only if local thread Glow Function is set.
+      if (localFusionInfo.function) {
+        // Fuse all linear operators.
+        FuseLinear(g);
+        // Fuse custom pass.
+        torch::jit::CustomFuseGraph(g, PyTorchModelLoader::isNodeSupported,
+                                    getFusionSymbol());
+      }
+    });
+  }
+};
+} // namespace
+
+/*static*/
+llvm::Error PyTorchFileLoader::loadPyTorchModel(
     const std::string &fileName,
-    std::shared_ptr<torch::jit::script::Module> &module, llvm::Error *errPtr) {
-  auto setup = [&]() -> llvm::Error {
-    try {
-      module = std::make_shared<torch::jit::script::Module>(
-          torch::jit::load(fileName));
-    } catch (const std::exception &x) {
-      RETURN_ERR(strFormat("Cannot load model from file: %s, , reason: %s",
-                           fileName.c_str(), x.what()));
+    std::shared_ptr<torch::jit::script::Module> &module) {
+  try {
+    module = std::make_shared<torch::jit::script::Module>(
+        torch::jit::load(fileName));
+  } catch (const std::exception &x) {
+    RETURN_ERR(strFormat("Cannot load model from file: %s, , reason: %s",
+                         fileName.c_str(), x.what()));
+  }
+  return llvm::Error::success();
+}
+
+/*static*/
+llvm::Error PyTorchFileLoader::loadPyTorchGraph(
+    const std::string &fileName, std::vector<torch::jit::IValue> &inputs,
+    glow::Function &F, std::vector<glow::Placeholder *> &inputPlaceholders,
+    std::vector<glow::Placeholder *> &outputPlaceholders, bool sanityCheck) {
+  // Register custom pass once.
+  static RegisterCustomFusionPass fuser;
+
+  // Convert PyTorch model into JIT Module.
+  std::shared_ptr<torch::jit::script::Module> module = {nullptr};
```

Comment:
nit: do you need to initialize a shared pointer like this? I think the default constructor is empty